source/geod24/LocalRest.d

```diff
@@ -303,22 +303,59 @@ public void sleep (Duration timeout)
         dg = If non-null, this delegate will be called when the timer fires
         periodic = Speficies if the timer fires repeatedly or only once
 
+    Returns:
+        A `Timer` instance with the ability to control the timer
+
 *******************************************************************************/
 
-public void setTimer (Duration timeout, void delegate() dg,
+public Timer setTimer (Duration timeout, void delegate() dg,
     bool periodic = false)
 {
     assert(scheduler !is null, "Cannot call this delegate from the main thread");
     assert(dg !is null, "Cannot call this delegate if null");
-    scheduler.schedule(
-    ()
+
+    Timer timer = new Timer(timeout, dg, periodic);
+    scheduler.schedule(() { timer.run(); });
```

Comment:
```suggestion
    scheduler.schedule(&timer.run);
```

This prevents an unneeded memory allocation.

Reply:
I applied suggestion.